### DISCUSSION TITLE
ruby-lsp not enabled in the gem's source files.

### DISCUSSION THREAD

**Discussion by edzhelyov** [Shopify/ruby-lsp#2701] [Q&A] [answered]
I'm using ruby-lsp as configured in the rails devcontainers. ruby-lsp is working in my project files where I can "Go to definition" in the source of different gems, but when in the specific gem source file I don't see "Go to definition" there and can't navigate around.

It seems ruby-lsp doesn't activate in these files, but I'm unable to find a solution.

Does anyone know how to enable "Go to definition", "Symbols" in the gem's source files?

**Comment by vinistock** [comment]
I suspect this might be a bug. We do register the [GEM_PATH](https://github.com/Shopify/ruby-lsp/blob/4fd4be9af67c1600e8ad1d762bc4ac87b74ec540/vscode/src/client.ts#L151) as directories for the language server to handle, but maybe the place where gems are being installed is overridden.

Do you happen to configure `BUNDLE_PATH` by any chance? Where are your gems installed relative to the project?

**Comment by edzhelyov** [comment]
Thank you, for taking the time to answer my question.

I'm using the default setup that comes with rails devcontainers in Rails 7.2. I checked around but I don't see BUNDLE_PATH set somewhere.

I tried with a fresh installation, and the issue is still present. I hope you can be able to recreate the steps, otherwise I will be happy to provide more information if you can guide me how to obtain it.

These are the steps that I follow:


> I don’t have any Ruby extensions installed in VScode, removed the Ruby LSP from my main installation.
rails new abc --devcontainer --skip-bundle

>rails -v
Rails 7.2.1

> Open VSCode and run the devcontainer. It will automatically run /bin/setup and we have the Ruby LSP extension installed, but the ruby-lsp server is not running.
Start with `Run Ruby LSP server` command in VSCode.

>Open a file from the rails project `application_cable/channel.rb` and then ‘Go to definition’ for ActionCable::Channel::Base, in the gem file Ruby LSP is not active, there is no options in the right click menu.

Gems are installed in the default ruby location under `/home/vscode/.rbenv` in the container.

**Comment by edzhelyov** [comment]
So I just tested outside of the container, directly on my machine, using Ruby 3.3.5 with rbenv and Rails 7.2.1. It does have the same issue, so it would seem it's not a devcontainer configuration.

**Comment by vinistock** [comment, marked as answer]
I believe I understood the problem. This is a bug for shim based version managers. I'm going to close this discussion so that we can move to the issue https://github.com/Shopify/ruby-lsp/issues/2709.

Thank you for reporting this!
